source/codegen/templates/service_helpers.mako

```diff
@@ -62,11 +62,16 @@ ${set_response_values(output_parameters)}\
 <%
   config = data['config']
   output_parameters = [p for p in parameters if common_helpers.is_output_parameter(p)]
+  session_parameter = parameters[0]['name']
 %>\
 ${initialize_input_params(function_name, parameters)}\
 ${initialize_output_params(output_parameters)}\
 % if function_name == config['close_function']:
       session_repository_->remove_session(${service_helpers.create_args(parameters)});
+% elif service_helpers.is_custom_close_method(function_data):
```

Comment:
See my comment below. Why do we need to treat custom close functions differently?

Reply:
@jonlugoatx the remove session takes session id as parameter. And calls the clean up lambda which was registered during InitExtCal function by passing id along with default parameters to clean up function. But for the case of custom close function, we don't want to always pass default values to library call. But instead pass on the values given by clients for those extra parameters

Reply:
Now I am treating close and custom close functions in same way